sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala

```diff
@@ -68,6 +68,16 @@ class ParquetFileFormat
 
   override def toString: String = "Parquet"
 
+  private def verifySchema(schema: StructType): Unit = {
+    if (schema.size == 0) {
+      throw new AnalysisException(
+        s"""
+           |Parquet data source does not support writing empty groups.
```

Comment:
`group` is a parquet term. Let's use `schema` instead?

We should also check for nested empty schema's.